lib/fbp_advanced.dart

```diff
@@ -243,26 +243,35 @@ class AdvancedFireBehaviourPredictionFormState
         CBH: _cbh,
         CFL: _cfl,
         HR: _minutes / 60.0);
-    // print('advanced: ${input}');
-    FireBehaviourPredictionPrimary prediction = FBPcalc(input, output: "ALL");
-    _onFMCChanged(prediction.FMC);
-    // Wind direction correction:
-    prediction.RAZ -= 180;
-    prediction.RAZ = prediction.RAZ < 0 ? prediction.RAZ + 360 : prediction.RAZ;
-    if (prediction.secondary != null) {
-      fireSize = getFireSize(
-          _fuelTypePreset!.code.name,
-          prediction.ROS,
-          prediction.secondary!.BROS,
-          _minutes,
-          prediction.CFB,
-          prediction.secondary!.LB);
+
+    FireBehaviourPredictionPrimary? prediction;
+    num surfaceFlameLength = 0;
+    int intensityClass = 1;
+    String? errorMessage;
+
+    try {
+      prediction = FBPcalc(input, output: "ALL");
+      // Wind direction correction:
+      prediction.RAZ -= 180;
```

Comment:
why does the 180 need to be subtracted?

Reply:
It's one of those confusing "are we talking about the direction something is coming from, or the direction something is going to" cases. In this case, say the direction of spread is North. That means that the net effective wind direction, is from the South. Hence, -180. You make a good point though - this should probably happen when the result is being pushed to the front end!

Reply:
Oh sweet - bug! Adjusting it here messed up something else. So I've changed how it works. Thanks.